core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -361,6 +363,27 @@ private String errorMsg(String operation) {
     return String.format("Operation %s is not supported after the table is serialized", operation);
   }
 
+  public static class SerializablePositionDeletesTable extends SerializableTable {
+
+    private final String baseTableName;
+
+    protected SerializablePositionDeletesTable(PositionDeletesTable deletesTable) {
+      super(deletesTable);
+      this.baseTableName = deletesTable.table().name();
+    }
+
+    @Override
+    protected Table newTable(TableOperations ops, String tableName) {
+      Table baseTable = new BaseTable(ops, baseTableName);
+      return new PositionDeletesTable(ops, baseTable, tableName);
+    }
+
+    @Override
+    public BatchScan newBatchScan() {
```

Comment:
Otherwise, we will go to the default which returns BatchScanAdapter

Reply:
Can't we override `newBatchScan()` in `SerializableTable`?